src/controllers/accomodation.controller.js

```diff
@@ -0,0 +1,62 @@
+import response from '../helpers/response.helper';
+import accomodation from '../services/accomodation.services';
+import UserServices from '../services/user.service';
+
+/**
+* Class for Travel Admins dealing with Accomodation
+*/
+class AccomdationController {
+  /**
+   * This function helps to to filter and store accomodation information
+   * @param { String } name name of accomodation
+   * @param { String } description description of accomodation
+   * @param { Integer } locationId location of accomodation
+   * @param { String } category category of accomodation
+   * @param { String } owner owner of accomodation
+   * @param { Array } images images of accomodation
+   * @param { Array } rooms rooms of accomodation
+   * @returns {Object} return created Accomodation
+   */
+  static async saveAccomodationInfo(name, description, locationId, category, owner, images, rooms) {
+    const newAccomodation = {
+      name, description, locationId, category, owner
+    };
+    const createdaccomodation = await accomodation.createAccomodation(newAccomodation);
+    const accomodationId = createdaccomodation.dataValues.id;
+    images.forEach(async ({ imageUrl }) => {
+      const newImage = { recordId: accomodationId, imageType: 'accomodation', imageUrl };
+      await accomodation.uploadImage(newImage);
+    });
+    rooms.forEach(async ({
+      nameRoom, typeId, price, currency, status
+    }) => {
+      const newRoom = {
+        name: nameRoom, accomodationId, typeId, price, status, currency
+      };
+      await accomodation.createRoom(newRoom);
+    });
+  }
+
+  /**
+   * This function helps to create new accomodation
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return user response
+   */
+  static async createAccomodation(req, res) {
+    const {
+      name, description, locationId, category, owner, images, rooms
+    } = req.body;
+    const user = await UserServices.findUserByEmail(req.user.email);
+    const role = user.role.toLowerCase();
+    if (role !== 'trip-administrator') {
+      response.errorMessage(res, 'You can not perform this Action', 401, 'error');
+    }
+    if (role === 'trip-administrator') {
+      const createdaccomodation = await AccomdationController.saveAccomodationInfo(name, description, locationId, category, owner, images, rooms);
```

Comment:
@Jajabenit250 It would be better to directly use services instead of calling another controller function.